### PR TITLE
Add $wgRevokePermissions to ManageWiki (T5991)

### PR DESCRIPTION
**Pull request by Universal-Omega**

### CODE REVIEW

File: includes/formFactory/ManageWikiFormFactoryBuilder.php [hunk 1]
```diff
@@ -579,6 +580,11 @@ private static function buildDescriptorPermissions(
 				'default' => wfMessage( 'managewiki-permissions-unassigned' )->text(),
 				'section' => 'unassigned'
 			],
+			'revoked' => [
+				'type' => 'info',
+				'default' => wfMessage( 'managewiki-permissions-revoked' )->text(),
```
Comment (RhinosF1) [inline includes/formFactory/ManageWikiFormFactoryBuilder.php:585]: Does this interface message exist?

**Review by RhinosF1 — CHANGES_REQUESTED** [review]

File: includes/formFactory/ManageWikiFormFactoryBuilder.php (hunk 1 above)
Comment (Universal-Omega) [reply includes/formFactory/ManageWikiFormFactoryBuilder.php:585]: Oh crap. I forgot to add that. Thanks. I'll do so now. I only know English so I'll only be able to add it to the English language though.

File: i18n/en.json [hunk 2]
```diff
@@ -136,6 +137,7 @@
 	"permissions-assigned": "Assigned Permissions",
 	"permissions-autopromote": "Autopromotion",
 	"permissions-unassigned": "Unassigned Permissions",
+	"permissions-revoked": "Revoked Permissions",
```
Comment (RhinosF1) [inline i18n/en.json:140]: Label for "Revoked Permissions" section

File: i18n/en.json [hunk 3]
```diff
@@ -85,6 +85,7 @@
 	"managewiki-permissions-addall": "Can add to others",
 	"managewiki-permissions-addself": "Can add to self",
 	"managewiki-permissions-assigned": "The below user rights are currently assigned to this user group.",
+	"managewiki-permissions-revoked": "Below you can explicitly revoke user rights of users to this group.",
```
Comment (RhinosF1) [inline i18n/en.json:88]: Description for "Revoked Permissions" tab

**Comment by RhinosF1** [comment]
See suggested qqq.json, nothing glares out but I'm not a php developer.

File: includes/formFactory/ManageWikiFormFactoryBuilder.php [hunk 4]
```diff
@@ -604,6 +610,17 @@ private static function buildDescriptorPermissions(
 			];
 		}
 
+		foreach ( $groupData['revokePermissions'] as $perm ) {
+			$formDescriptor["right-{$perm}"] = [
+				'type' => 'check',
+				'label' => $perm,
+				'help' => User::getRightDescription( $perm ),
+				'section' => revoked,
```
Comment (supertassu) [inline includes/formFactory/ManageWikiFormFactoryBuilder.php:618]: Missing quotes

File: sql/patches/patch-revoke-permissions.sql [hunk 5]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE /*$wgDBprefix*/mw_permissions
+  ADD COLUMN perm_revoke LONGTEXT AFTER perm_autopromote;
```
Comment (supertassu) [inline sql/patches/patch-revoke-permissions.sql:2]: `AFTER` does not match with `mw_permissions.sql`

File: sql/patches/patch-revoke-permissions.sql [hunk 6]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE /*$wgDBprefix*/mw_permissions
```
Comment (supertassu) [inline sql/patches/patch-revoke-permissions.sql:1]: I believe this file should be registered in `ManageWikiHooks::fnManageWikiSchemaUpdates`

**Review by supertassu — CHANGES_REQUESTED** [review]

File: sql/patches/patch-revoke-permissions.sql (hunk 5 above)
Comment (Universal-Omega) [reply sql/patches/patch-revoke-permissions.sql:2]: How so?

Comment (supertassu) [reply sql/patches/patch-revoke-permissions.sql:2]: On `mw_permissions.sql` `perm_revoke` was added after `perm_permissions` but here it's added after `perm_autopromote`

Comment (Universal-Omega) [reply sql/patches/patch-revoke-permissions.sql:2]: Oh, crap your right. Thanks for pointing that out!

File: sql/patches/patch-revoke-permissions.sql (hunk 6 above)
Comment (Universal-Omega) [reply sql/patches/patch-revoke-permissions.sql:1]: Fixed

**Comment by Universal-Omega** [comment]
I believe this has now been finished and should theoretically work. I re-reviewed it again to make sure and it all now looks good.

**Pull request closed without merging**